source/material_model/rheology/strain_dependent.cc

```diff
@@ -425,6 +427,8 @@ namespace aspect
                 else
                   out.reaction_terms[i][this->introspection().compositional_index_for_name("viscous_strain")] = e_ii;
               }
+            if (this->introspection().compositional_name_exists("noninitial_plastic_strain") && plastic_yielding == true && weakening_mechanism == plastic_weakening_with_plastic_strain_only)
```

Comment:
Why the third condition? Should this not be happening whenever plastic yielding happens, even if it is not weakening the material?
If you change the condition remember to also update the documentation change above.

Reply:
@gassmoeller - the idea behind the third condition was to make sure there was no confusion about whether strain weakening was being used even if  `strain weakening mechanism = none`. With said, anyone who uses this option will certainly have carefully read through the documentation and knows exactly what the field represents. I've updated accordingly.